public-ui/src/app/data-browser/views/survey-view/survey-react-view.component.tsx

```diff
@@ -0,0 +1,466 @@
+                                      const a = new Date(a1.year, a1.monthNum.split('/')[0], 1);
+                                      const b = new Date(a2.year, a2.monthNum.split('/')[0], 1);
+                                      return a.valueOf() - b.valueOf();
+                                  });
+                                  this.setState({surveyVersions: surveyVersions});
+                        }
+                );
+            }
+        this.setState({survey: survey, surveyPdfUrl: surveyPdfUrl, isCopeSurvey: copeFlag, extraQuestionConceptIds: extraConcepts}, () => {
+            this.getSurvey(); });
+    }
+  }
+
+  handleChange(val) {
+        this.setState({ searchWord: val });
+        this.searchCount(val);
+        this.search(val);
+  }
+
+  getSurvey() {
+        const {survey, searchWord, extraQuestionConceptIds} = this.state;
+        api.getSurveyQuestions(survey.conceptId.toString(), searchWord, extraQuestionConceptIds).then(
+            (x: any) => {
+                this.processSurveyQuestions(x);
+        });
+  }
+
+  processSurveyQuestions(results: any) {
+        const survey = results.survey;
+        const questions = results.questions.items;
+        this.setDefaults(questions, 0, survey);
+  }
+
+  setDefaults(surveyQuestions: any, level: any, survey) {
+        const {isCopeSurvey} = this.state;
+        for (const q of surveyQuestions) {
+            q.actualQuestionNumber = q.questionOrderNumber;
+            if (isCopeSurvey) {
+                q.graphToShow = GraphType.SurveyVersion;
+                q.selectedAnalysis = q.versionAnalysis;
+            } else {
+                q.graphToShow = GraphType.BiologicalSex;
+                q.selectedAnalysis = q.genderAnalysis;
+            }
+            q.graphDataToShow = 'Count';
+            q.resultFetchComplete = false;
+        }
+        surveyQuestions.sort((a1, a2) => {
+            if (a1.actualQuestionNumber < a2.actualQuestionNumber) {
+                return -1;
+            }
+            if (a1.actualQuestionNumber > a2.actualQuestionNumber) {
+                return 1;
+            }
+            return 0;
+        });
+        this.setState({questions: surveyQuestions, survey: survey, loading: false});
+  }
+
+  backToMain() {
+        navigateByUrl('');
+  }
+
+
+  render() {
+        const {loading, searchWord, isCopeSurvey, survey, questions, surveyVersions, surveyPdfUrl, showStatement} = this.state;
+        const statClass = isCopeSurvey ? 'cope-stat-layout' : 'stat-layout';
+        const statStyle = isCopeSurvey ? styles.copeStatLayout : styles.statLayout;
+        return <React.Fragment>
+        <style>{surveyStyle}</style>
+        <div className='survey-view' style={styles.surveyView}>
+        {survey && <SurveyDescReactComponent surveyName={survey.name} isCopeSurvey={isCopeSurvey} surveyDescription={survey.description}
+        click={this.showCopeStatement}/>
```

Comment:
We should set state inline here instead of calling a separate function. Then we can remove the `showCopeStatement` function:
```suggestion
        click={() => this.setState({showStatement: true})}/>
```

Reply:
Changed this thanks.